src/fsharp/IlxGen.fs

```diff
@@ -2201,150 +2208,169 @@ and GenExprWithStackGuard cenv cgbuf eenv sp expr sequel =
     | :? System.InsufficientExecutionStackException ->
         error(InternalError(sprintf "Expression is too large and/or complex to emit. Method name: '%s'. Recursive depth: %i." cgbuf.MethodName cenv.exprRecursionDepth, expr.Range))
 
-and GenExprAux (cenv: cenv) (cgbuf: CodeGenBuffer) eenv sp expr sequel =
-  let g = cenv.g
-  let expr = stripExpr expr
+/// Process the debug point and check for alternative ways to generate this expression.
+/// Returns 'true' if the expression was processed by alternative means.
+and GenExprPreSteps (cenv: cenv) (cgbuf: CodeGenBuffer) eenv sp expr sequel =
```

Comment:
This factors out these "pre steps" into a separate method, which we use from both GenExprAux and GenLinearExpr.

This factoring is useful for me in the other PR when detecting and generating code for state machines.